test/e2e/case17_gatekeeper_sync_test.go

```diff
@@ -251,6 +251,22 @@ var _ = Describe("Test Gatekeeper ConstraintTemplate and constraint sync", Order
 			)
 		}, gkAuditFrequency*3, 1).Should(Succeed())
 
+		By("Checking if policy status history is correctly formatted")
```

Comment:
Without the code change, I would think this test would still pass because the issue seems to be in the template-sync's event emitting as opposed to the Gatekeeper sync controller. The original bug report mentioned Gatekeeper but that error actually originated from the template-sync when it was trying to create the Gatekeeper constraint from the `policy-templates` array.

I would adjust the existing test in case10 to make it more precise.